mapper/mapper.go

```diff
@@ -420,68 +425,56 @@ func CreateAgePage(req *http.Request, f filter.Model, d dataset.DatasetDetails,
 
 	p.Data.FormAction.URL = fmt.Sprintf("/filters/%s/dimensions/age/update", f.FilterID)
 
-	var ages []int
+	// get mapping of labels (keys) to options (values) and initialise aux vars
 	labelIDs := getNameIDLookup(allVals)
-	for _, age := range allVals.Items {
-		ageInt, err := strconv.Atoi(age.Label)
-		if err != nil {
-			if strings.Contains(age.Label, "+") {
-				p.Data.Oldest = age.Label
-			} else {
+	youngest := math.MaxInt32
+	oldest := math.MinInt32
+
+	// iterate all values, and add them to the Page in the same order,
+	// settign the 'isSelected' for each one of them (according to selVals)
+	// and setting oldest and youngest values
+	for _, a := range allVals.Items {
```

Comment:
Note:
I had to refactor this loop because the original code was setting the oldest value (if it contained '+') at the end of the list, regardless of the order returned by the API.